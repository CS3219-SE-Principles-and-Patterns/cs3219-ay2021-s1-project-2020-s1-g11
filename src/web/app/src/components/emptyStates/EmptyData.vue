<template>
  <div class="emptySection">
    <el-row>
      <el-col>
        <div class="icon">
          <img src="@/assets/businessman.png" />
        </div>
      </el-col>
    </el-row>
    <el-row>
      <el-col>
        <p>
          <i> No data found. </i>
        </p>
        <p>
          Let's get started by importing data.
        </p>
        <el-button type="primary" icon="el-icon-plus" @click="importData">Import Data</el-button>
      </el-col>
    </el-row>
  </div>
</template>

<script>
export default {
  methods: {
    importData() {
      this.$router.push("/importData");
    },
  }
}
</script>

<style scoped>
.icon {
  margin-left: 04.5rem;
  margin-bottom: 0.25rem;
}
.emptySection {
  text-align: center;
  vertical-align: middle;
}
</style>
