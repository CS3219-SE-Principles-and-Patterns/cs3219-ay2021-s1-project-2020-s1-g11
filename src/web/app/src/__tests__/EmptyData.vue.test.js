import {createLocalVue, shallowMount} from '@vue/test-utils'
import ElementUI from 'element-ui';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import EmptyData from "./../components/emptyStates/EmptyData.vue";

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);
localVue.use(ElementUI);

describe("EmptyData.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        isPageLoading: false,
        userInfo: {
          isApiError: false,
        },
        dbMetaData: {
          entitiesStatus: {
            isLoading: false,
          }
        }
      }
    })
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(EmptyData, {store, localVue})
    expect(wrapper.element).toMatchSnapshot()
  });
});