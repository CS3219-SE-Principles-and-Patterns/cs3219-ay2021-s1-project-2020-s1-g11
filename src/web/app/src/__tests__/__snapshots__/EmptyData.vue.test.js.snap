// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyData.vue renders correctly 1`] = `
<div
  class="emptySection"
>
  <el-row-stub
    align="top"
    justify="start"
    tag="div"
  >
    <el-col-stub
      span="24"
      tag="div"
    >
      <div
        class="icon"
      >
        <img
          src="@/assets/businessman.png"
        />
      </div>
    </el-col-stub>
  </el-row-stub>
   
  <el-row-stub
    align="top"
    justify="start"
    tag="div"
  >
    <el-col-stub
      span="24"
      tag="div"
    >
      <p>
        <i>
           No data found. 
        </i>
      </p>
       
      <p>
        
        Let's get started by importing data.
      
      </p>
       
      <el-button-stub
        icon="el-icon-plus"
        nativetype="button"
        type="primary"
      >
        Import Data
      </el-button-stub>
    </el-col-stub>
  </el-row-stub>
</div>
`;
